compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1968,6 +1968,24 @@ object messages {
     }
   }
 
+  case class BadSymbolicReference(
+    location: String,
+    name: String,
+    denotationOwner: Symbol,
+    src: String
+  )(implicit ctx: Context) extends Message(BadSymbolicReferenceID) {
```

Comment:
I would change `BadSymbolicReference ` so that it only has `denot: SymDenotation`. You can derive `location`, `name`, `owner` and `src` from `denot`.